### PR TITLE
SageMaker DP Support

### PR DESCRIPTION
**Pull request by pacman100**
### What does this PR do?
1. Fixes bugs in the current SageMaker support to get it working properly using the latest HF DLC
2. Adds `DATA_PARALLEL` support. Run `accelerate config` and answer the questions and choose `DATA_PARALLEL` for SageMaker Distribution Type. Sample config below with `XXXXX` values being AWS account specific.
```bash
base_job_name: accelerate-sagemaker-1
compute_environment: AMAZON_SAGEMAKER
distributed_type: DATA_PARALLEL
ec2_instance_type: ml.p3.16xlarge
iam_role_name: XXXXX
mixed_precision: fp16
num_machines: 1
profile: XXXXX
py_version: py38
pytorch_version: 1.10.2
region: us-east-1
transformers_version: 4.17.0
use_cpu: false
```
After above config, just run the below command to run the official NLP example:
```bash
cd acceelerate/examples
accelerate launch complete_nlp_example.py
```
The output logs:

Screenshot 2022-07-08 at 1 55 52 AM

### CODE REVIEW

File: src/accelerate/state.py [hunk 1]
```diff
@@ -89,7 +90,27 @@ def __init__(
...
+                        torch.distributed.init_process_group(backend="smddp")
+                    self.backend = "smddp"
+                    self.num_processes = torch.distributed.get_world_size()
+                    self.process_index = torch.distributed.get_rank()
+                    self.local_process_index = int(os.environ.get("LOCAL_RANK", -1))
+                    self.device = torch.device("cuda", self.local_process_index)
+                    torch.cuda.set_device(self.device)
+                    self.mixed_precision = mixed_precision
+                elif os.environ.get("SAGEMAKER_DISTRIBUTED_TYPE") == SageMakerDistributedType.MODEL_PARALLEL:
+                    raise NotImplementedError("Model Parallelism is not implemented yet. We are working on it")
```
Comment (sgugger) [inline src/accelerate/state.py:112]: Would remove the choice and error message for now as I don't think we will add this any time soon :-)

**Review by sgugger — APPROVED** [review]
Thanks for working on this! Let's also see what @philschmid thinks since he worked on the sagemaker command in Accelerate.

**Comment by plamb-viso** [comment]
I will test this with my use case as soon as it gets merged into `main`

**Review by philschmid — APPROVED** [review]
LGTM! How should we maintain the `SAGEMAKER_PYTHON_VERSION, SAGEMAKER_PYTORCH_VERSION, SAGEMAKER_TRANSFORMERS_VERSION` versions once we release new DLCs?

**Comment by pacman100** [comment]
> LGTM! How should we maintain the `SAGEMAKER_PYTHON_VERSION, SAGEMAKER_PYTORCH_VERSION, SAGEMAKER_TRANSFORMERS_VERSION` versions once we release new DLCs?

Hello, as it is not a frequent change, it can be done manually for time being whenever new releases of DLC happen. Any suggestions/best practices for automating it?

**Comment by HebaGamalElDin** [comment]
Hello @pacman100, what is the sagemaker SDK estimator supposed to be used with accelerate?

**Pull request merged**